pkgs/development/python-modules/bottleneck/default.nix

```diff
@@ -21,8 +21,12 @@ buildPythonPackage rec {
   '';
 
   checkInputs = [ pytest nose ];
+  # test_make_c_files expect a missing test data file, therefore, the test fails.
   checkPhase = ''
-    py.test -p no:warnings $out/${python.sitePackages}
+    py.test \
```

Comment:
This is a good opportunity to switch this over to pytestCheckHook.

Reply:
Tried, it fails because well:
```
============================= test session starts ==============================
platform linux -- Python 3.9.11, pytest-7.0.1, pluggy-1.0.0
rootdir: /build/Bottleneck-1.3.4, configfile: setup.cfg
collected 0 items / 9 errors                                                   

==================================== ERRORS ====================================
_________ ERROR collecting bottleneck/tests/input_modification_test.py _________
ImportError while importing test module '/build/Bottleneck-1.3.4/bottleneck/tests/input_modification_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/bvzpppawf2naqv3qhxqv66jxaq3iaxyj-python3-3.9.11/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
bottleneck/__init__.py:7: in <module>
    from .move import (move_argmax, move_argmin, move_max, move_mean, move_median,
E   ModuleNotFoundError: No module named 'bottleneck.move'
_____________ ERROR collecting bottleneck/tests/list_input_test.py _____________
ImportError while importing test module '/build/Bottleneck-1.3.4/bottleneck/tests/list_input_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/bvzpppawf2naqv3qhxqv66jxaq3iaxyj-python3-3.9.11/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
bottleneck/tests/list_input_test.py:7: in <module>
    import bottleneck as bn
bottleneck/__init__.py:7: in <module>
    from .move import (move_argmax, move_argmin, move_max, move_mean, move_median,
E   ModuleNotFoundError: No module named 'bottleneck.move'
_______________ ERROR collecting bottleneck/tests/memory_test.py _______________
ImportError while importing test module '/build/Bottleneck-1.3.4/bottleneck/tests/memory_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/bvzpppawf2naqv3qhxqv66jxaq3iaxyj-python3-3.9.11/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
bottleneck/tests/memory_test.py:3: in <module>
    import bottleneck as bn
bottleneck/__init__.py:7: in <module>
    from .move import (move_argmax, move_argmin, move_max, move_mean, move_median,
E   ModuleNotFoundError: No module named 'bottleneck.move'
________________ ERROR collecting bottleneck/tests/move_test.py ________________
ImportError while importing test module '/build/Bottleneck-1.3.4/bottleneck/tests/move_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/bvzpppawf2naqv3qhxqv66jxaq3iaxyj-python3-3.9.11/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
bottleneck/tests/move_test.py:5: in <module>
    import bottleneck as bn
bottleneck/__init__.py:7: in <module>
    from .move import (move_argmax, move_argmin, move_max, move_mean, move_median,
E   ModuleNotFoundError: No module named 'bottleneck.move'
___________ ERROR collecting bottleneck/tests/nonreduce_axis_test.py ___________
ImportError while importing test module '/build/Bottleneck-1.3.4/bottleneck/tests/nonreduce_axis_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/bvzpppawf2naqv3qhxqv66jxaq3iaxyj-python3-3.9.11/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
bottleneck/tests/nonreduce_axis_test.py:9: in <module>
    import bottleneck as bn
bottleneck/__init__.py:7: in <module>
    from .move import (move_argmax, move_argmin, move_max, move_mean, move_median,
E   ModuleNotFoundError: No module named 'bottleneck.move'
_____________ ERROR collecting bottleneck/tests/nonreduce_test.py ______________
ImportError while importing test module '/build/Bottleneck-1.3.4/bottleneck/tests/nonreduce_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/bvzpppawf2naqv3qhxqv66jxaq3iaxyj-python3-3.9.11/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
bottleneck/tests/nonreduce_test.py:7: in <module>
    import bottleneck as bn
bottleneck/__init__.py:7: in <module>
    from .move import (move_argmax, move_argmin, move_max, move_mean, move_median,
E   ModuleNotFoundError: No module named 'bottleneck.move'
_______________ ERROR collecting bottleneck/tests/reduce_test.py _______________
ImportError while importing test module '/build/Bottleneck-1.3.4/bottleneck/tests/reduce_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/bvzpppawf2naqv3qhxqv66jxaq3iaxyj-python3-3.9.11/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
bottleneck/tests/reduce_test.py:9: in <module>
    import bottleneck as bn
bottleneck/__init__.py:7: in <module>
    from .move import (move_argmax, move_argmin, move_max, move_mean, move_median,
E   ModuleNotFoundError: No module named 'bottleneck.move'
____________ ERROR collecting bottleneck/tests/scalar_input_test.py ____________
ImportError while importing test module '/build/Bottleneck-1.3.4/bottleneck/tests/scalar_input_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/bvzpppawf2naqv3qhxqv66jxaq3iaxyj-python3-3.9.11/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
bottleneck/tests/scalar_input_test.py:4: in <module>
    import bottleneck as bn
bottleneck/__init__.py:7: in <module>
    from .move import (move_argmax, move_argmin, move_max, move_mean, move_median,
E   ModuleNotFoundError: No module named 'bottleneck.move'
______________ ERROR collecting bottleneck/tests/test_template.py ______________
ImportError while importing test module '/build/Bottleneck-1.3.4/bottleneck/tests/test_template.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/bvzpppawf2naqv3qhxqv66jxaq3iaxyj-python3-3.9.11/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
bottleneck/tests/test_template.py:4: in <module>
    from ..src.bn_template import make_c_files
bottleneck/__init__.py:7: in <module>
    from .move import (move_argmax, move_argmin, move_max, move_mean, move_median,
E   ModuleNotFoundError: No module named 'bottleneck.move'
=========================== short test summary info ============================
ERROR bottleneck/tests/input_modification_test.py
ERROR bottleneck/tests/list_input_test.py
ERROR bottleneck/tests/memory_test.py
ERROR bottleneck/tests/move_test.py
ERROR bottleneck/tests/nonreduce_axis_test.py
ERROR bottleneck/tests/nonreduce_test.py
ERROR bottleneck/tests/reduce_test.py
ERROR bottleneck/tests/scalar_input_test.py
ERROR bottleneck/tests/test_template.py
!!!!!!!!!!!!!!!!!!! Interrupted: 9 errors during collection !!!!!!!!!!!!!!!!!!!!
============================== 9 errors in 0.59s ===============================
```


We are in presence of an external native module, it seems to bring some unexpected things to the hook.

Reply:
Try adding

```
preCheck = ''
  export PYTHONPATH=$out/${python.sitePackages}:$PYTHONPATH
'';

Reply:
Same issues, @SuperSandro2000.

Reply:
Then I don't know